src/idl_gen_cpp.cpp

```diff
@@ -2443,9 +2488,15 @@ class CppGenerator : public BaseGenerator {
               // the underlying storage type (eg. uint8_t).
               const auto basetype = GenTypeBasic(
                   field.value.type.enum_def->underlying_type, false);
-              code += "_fbb.CreateVectorScalarCast<" + basetype +
-                      ">(flatbuffers::data(" + value + "), " + value +
-                      ".size())";
+              code += "_fbb.CreateVectorScalarCast<" + basetype + ">";
+              if (!NativeVectorDefined(&field)) {
+                // Use std::vector variant, requires flatbuffers::data() for
```

Comment:
Ok, now we have a function, but not all code is using it, that is almost worse than not having a function. This code is getting very confusing to follow.